tests/management/access/test_view.py

```diff
@@ -138,3 +140,66 @@ def test_missing_invalid_username(self):
         client = APIClient()
         response = client.get(url, **self.headers)
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_admin_RonR(self):
+        """Test that a admin user can access RBAC resources"""
+        url = '{}?application={}&username={}'.format(reverse('access'), 'rbac', self.principal.username)
+        client = APIClient()
+        response = client.get(url, **self.headers)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+
+class AccessViewNonAdminTests(IdentityRequest):
+    """Test the access view for nonadmin user."""
+
+    def setUp(self):
+        """Set up the access view nonadmin tests."""
+        super().setUp()
+
+        self.user_data = self._create_user_data()
+        self.customer = self._create_customer_data()
+        self.request_context = self._create_request_context(self.customer,
+                                                            self.user_data,
+                                                            is_org_admin=False)
+
+
+        request = self.request_context['request']
+        self.headers = request.META
+        self.access_data = {
+            'permission': 'app:*:*',
+            'resourceDefinitions': [
+                {
+                    'attributeFilter': {
+                        'key': 'key1',
+                        'operation': 'equal',
+                        'value': 'value1'
+                    }
+                }
+            ]
+        }
+        with tenant_context(self.tenant):
+            self.principal = Principal(username=self.user_data['username'])
+            self.principal.save()
+            self.admin_principal = Principal(username="user_admin")
+            self.admin_principal.save()
+            self.group = Group(name='groupA')
+            self.group.save()
+            self.group.principals.add(self.principal)
+            self.group.save()
+
+    def tearDown(self):
+        """Tear down access view tests."""
+        User.objects.all().delete()
+        with tenant_context(self.tenant):
+            Group.objects.all().delete()
+            Principal.objects.all().delete()
+            Role.objects.all().delete()
+            Policy.objects.all().delete()
+
+    def test_nonadmin_RonR(self):
+        """Test that a nonadmin user can't access RBAC resources"""
+        url = '{}?application={}&username={}'.format(reverse('access'), 'rbac', self.principal.username)
```

Comment:
This is the one I'd expect to still be a 200 as a non-admin, so I must be mistaken in how I'm interpreting these tests. If I test the access endpoint locally as a non-admin, I get a 200 with an empty set for RBAC as expected, but this appears to 403? If I check the principals, groups, or roles endpoints I do get a 403 which I would also expect.

Reply:
@coderbydesign Do you mean when a non-admin want to get access for himself, it is reasonable to return 200 with empty set.
But when he wants to get access for others, it should be returning forbidden?

Reply:
Yeah. Is that what's happening? The access query isn't for the principal on the request? That would make sense.

Reply:
Agree 👍